notification/slack/channel.go

```diff
@@ -327,30 +331,52 @@ func (s *ChannelSender) Send(ctx context.Context, msg notification.Message) (*no
 			break
 		}
 
-		vals.Set("text", fmt.Sprintf("Alert: %s\n\n<%s>", t.Summary, cfg.CallbackURL("/alerts/"+strconv.Itoa(t.AlertID))))
+		a.Summary = t.Summary
+		a.ID = t.AlertID
+		a.Status = alert.StatusTriggered
+
+		msgOpt := CraftAlertMessage(a, cfg.CallbackURL("/alerts/"+strconv.Itoa(a.ID)), "")
+		_, ts, _, err := api.SendMessageContext(ctx, msg.Destination().Value, msgOpt...)
+		if err != nil {
+			return nil, err
+		}
+		return &notification.SentMessage{
+			ExternalID: ts,
+			State:      notification.StateDelivered,
+			SrcValue:   "",
+		}, nil
 	case notification.AlertStatus:
-		vals.Set("thread_ts", t.OriginalStatus.ProviderMessageID.ExternalID)
-		var status string
-		switch t.NewAlertStatus {
-		case alert.StatusActive:
-			status = "Acknowledged"
-		case alert.StatusTriggered:
-			status = "Unacknowledged"
-		case alert.StatusClosed:
-			status = "Closed"
+		_a, err := s.cfg.AlertStore.FindOne(ctx, t.AlertID)
+		if err != nil {
+			return nil, err
 		}
+		a = *_a
 
-		text := "Status Update: " + status + "\n" + t.LogEntry
-		vals.Set("text", text)
+		msgOpt := CraftAlertMessage(a, cfg.CallbackURL("/alerts/"+strconv.Itoa(a.ID)), "")
+		_, ts, _, err := api.UpdateMessageContext(ctx, msg.Destination().Value, t.Dest.Value, msgOpt...)
+		if err != nil {
+			return nil, err
+		}
+		vals.Set("thread_ts", t.OriginalStatus.ProviderMessageID.ExternalID)
+		// todo, post to thread
+		return &notification.SentMessage{
+			ExternalID: ts,
+			State:      notification.StateDelivered,
+			SrcValue:   "",
```

Comment:
SrcValue should be the bot user ID

Reply:
it doesn't seem like i have access to the bot ID from the result using the library. what is this used for? iirc we didn't need this value for anything on the first pass.